File: src/services/history.ts
import axios from 'axios';
import { API_BASE_URL } from '../config';

export interface HistoryItem {
  id: string;
  type: 'document'|'chat'|'login'|'upload'|'approval';
  title: string;
  description: string;
  timestamp: string;    // ISO 字符串
  status?: 'success'|'error'|'pending';
  user?: string;
  // details 根据后端返回而定
  details?: {
    fileSize?: string;
    duration?: string;
    result?: string;
  };
}

interface ListParams {
  start?: string;  // ISO
  end?: string;
  type?: string;
  query?: string;
}

export async function fetchHistory(params: ListParams = {}): Promise<HistoryItem[]> {
  const res = await axios.get<HistoryItem[]>(`${API_BASE_URL}/history`, { params });
  return res.data;
}

File: app/models/history.py
# backend/app/models/history.py

from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class HistoryItem(BaseModel):
    id: str
    type: str               # 'document'|'chat'|'login'|'upload'|'approval'
    title: str
    description: str
    timestamp: datetime
    status: Optional[str] = None
    user: Optional[str] = None
    # 如果还有 details 字段，也可以加：
    # fileSize: Optional[str] = None
    # duration: Optional[str] = None
